src/pages/index.js

```diff
@@ -117,32 +165,92 @@ function handleCloseModal(modal) {
 }
 
+function returnToSave(button) {
+  setTimeout(() => {
+    button.textContent = "Save";
+  }, 2000);
+}
+
 function handleProfileFormSubmit(evt) {
   evt.preventDefault();
-  handleCloseModal(profileFormElement);
+  profileSubmitButton.textContent = "Saving...";
+  const patch = { name: nameInput.value, about: jobInput.value };
+  api
+    .editUserInfo(patch)
+    .then((data) => {
+      profileNameElement.textContent = data.name;
+      profileJobElement.textContent = data.about;
+      handleCloseModal(profileFormElement);
+      returnToSave(profileSubmitButton);
+    })
+    .catch((err) => console.error(err));
 }
 
-function renderCards() {
-  initialCards.forEach((card) => {
-    cardsListElement.append(getCardElementData(card));
-  });
+function handleNewPostSubmit(evt) {
+  evt.preventDefault();
+  newPostSubmitButton.textContent = "Saving...";
+  api
+    .postNewCard(captionInput.value, linkInput.value)
+    .then((res) => {
+      const cardElement = getCardElementData(res);
+      cardsListElement.prepend(cardElement);
+      evt.target.reset();
+      disableButton(newPostSubmitButton, settings);
+      handleCloseModal(newPostFormElement);
+      returnToSave(newPostSubmitButton);
+    })
+    .catch((err) => {
+      console.log("Error creating card:", err);
+    });
 }
 
-function renderNewCard(card) {
-  cardsListElement.prepend(getCardElementData(card));
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  avatarSubmitButton.textContent = "Saving...";
+  api
+    .editUserAvatar(avatarInput.value)
+    .then((data) => {
+      profileImg.src = data.avatar;
+      handleCloseModal(avatarModal);
+      returnToSave(avatarSubmitButton);
+      avatarInput.value = "";
+    })
+    .catch((err) => console.error(err));
 }
 
-function handleNewPostSubmit(evt) {
+function handleDeleteSubmit(evt) {
   evt.preventDefault();
-  const newPostElement = {
-    link: linkInput.value,
-    name: captionInput.value,
-  };
-  initialCards.unshift(newPostElement);
-  renderNewCard(newPostElement);
-  evt.target.reset();
-  disableButton(newPostSubmitButton, settings);
-  handleCloseModal(newPostFormElement);
+  api
+    .removeCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      handleCloseModal(deleteModal);
+    })
+    .catch((err) => console.error(err));
+}
+
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  handleOpenModal(deleteModal);
+}
+
+function handleLikeClick(cardElement, btnElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  let selectedBtn = btnElement;
+
+  if (!selectedBtn.classList.contains("card__like-btn_liked")) {
+    api
+      .likeCard(selectedCardId)
+      .then(selectedBtn.classList.add("card__like-btn_liked"))
+      .catch((err) => console.error(err));
+  } else {
+    api
+      .removeLike(selectedCardId)
+      .then(selectedBtn.classList.remove("card__like-btn_liked"))
```

Comment:
`      .then(selectedBtn.classList.remove("card__like-btn_liked"))
`

In `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 


```js

      .then(() => selectedBtn.classList.remove("card__like-btn_liked"))

```